src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java

```diff
@@ -56,6 +68,82 @@ public synchronized String getCurrentCommandString() {
 			return getCurrentCommand().toString();
 	}
 
+	/**
+	 * Runs current command, if set.
+	 *
+	 */
+
+	public synchronized void runCurrentCommand(){
+		if(currentCommand != null){
+			currentCommand.execute(driverManager.getCurrentDriver());
+		}
+	}
+
+	/**
+	 * Deletes observers and move observers collection to cache
+	 */
+
+	public synchronized void deleteCurrentObservers(){
+
+		if(changePublisher.getSubscribers().size() > 0){
+			cache = new ArrayList<>(changePublisher.getSubscribers());
+		}
+
+		changePublisher.clearObservers();
+	}
+
+	/**
+	 * Restores observers from cache
+	 * NOTE:
+	 * It's probable if addObserver/s feature would be implemented, the cache should be empty
+	 */
+
+	public synchronized void resetObservers(){
+		if(cache != null && cache.size() > 0){
+			cache.forEach(cached -> changePublisher.addSubscriber(cached));
```

Comment:
Isn't it a duplication of addObservers method code?

Reply:
You're right, I changed it to addSubscribers(cache) call instead of that duplication in latest commit.